nltk/parse/malt.py

```diff
@@ -5,245 +6,492 @@
+	"""
+	A class for dependency parsing with MaltParser. The input is the paths to:
+     - a maltparser directory
+     - (optionally) the path to a pre-trained MaltParser .mco model file
+     - (optionally) the tagger to use for POS tagging before parsing
+     - (optionally) additional Java arguments
+	
+	Example:
+        >>> from nltk.parse import malt
+        >>> # With MALT_PARSER and MALT_MODEL environment set.
+        >>> mp = malt.MaltParser('maltparser-1.7.2', 'engmalt.linear-1.7.mco') # doctest: +SKIP
+        >>> mp.parse_one('I shot an elephant in my pajamas .'.split()).tree() # doctest: +SKIP
+        (shot I (elephant an) (in (pajamas my)) .)
+        >>> # Without MALT_PARSER and MALT_MODEL environment.
+        >>> mp = malt.MaltParser('/home/alvas/maltparser-1.7.2/', '/home/alvas/engmalt.linear-1.7.mco') # doctest: +SKIP
+        >>> mp.parse_one('I shot an elephant in my pajamas .'.split()).tree() # doctest: +SKIP
+        (shot I (elephant an) (in (pajamas my)) .)
+	"""
+	def __init__(self, parser_dirname, model_filename=None, tagger=None, 
+				 additional_java_args=[]):
+		"""
+		An interface for parsing with the Malt Parser.
+
+		:param parser_dirname: The path to the maltparser directory that 
+		contains the maltparser-1.x.jar
+		:type parser_dirname: str
+		:param model_filename: The name of the pre-trained model with .mco file 
+		extension. If provided, training will not be required.
+		(see http://www.maltparser.org/mco/mco.html and 
+		see http://www.patful.com/chalk/node/185)
+		:type model_filename: str
+		:param tagger: The tagger used to POS tag the raw string before 
+		formatting to CONLL format. It should behave like `nltk.pos_tag`
+		:type tagger: function
+		:param additional_java_args: This is the additional Java arguments that 
+		one can use when calling Maltparser, usually this is the heapsize 
+		limits, e.g. `additional_java_args=['-Xmx1024m']`
+		(see http://goo.gl/mpDBvQ)
+		:type additional_java_args: list
+		"""
+		
+		# Find all the necessary jar files for MaltParser.
+		self.malt_jars = find_maltparser(parser_dirname)
+		# Initialize additional java arguments.
+		self.additional_java_args = additional_java_args
+		# Initialize model.
+		self.model = find_malt_model(model_filename)
+		self._trained = False if self.model == 'malt_temp.mco' else True
+		# Set the working_dir parameters i.e. `-w` from MaltParser's option.
+		self.working_dir = tempfile.gettempdir() 
+		# Initialize POS tagger.
+		if tagger is not None:
+			self.tagger = tagger
+		else:
+			self.tagger = malt_regex_tagger()	
+
+	def pretrained_model_sanity_checks(self, tree_str):
+		"""
+		Performs sanity checks and replace oddities in pre-trained model
+		outputs from http://www.maltparser.org/mco/english_parser/engmalt.html
+		
+		:param tree_str: The CONLL output file for a single parse
+		:type tree_str: str
+		:return: str
+		"""
+		# Checks for oddities in English pre-trained model.
+		if '\t0\tnull\t' in tree_str and \
```

Comment:
I would rewrite this to avoid `\`
```python
if (
    '\t0\tnull\t' in tree_str
     and self.model.endswith(('engmalt.linear-1.7.mco', 'engmalt.poly-1.7.mco'))
):
```